src/context/WalletProvider/WalletProvider.tsx

```diff
@@ -83,15 +94,21 @@ const reducer = (state: InitialState, action: ActionTypes) => {
     case WalletActions.SET_ADAPTERS:
       return { ...state, adapters: action.payload }
     case WalletActions.SET_WALLET:
-      return {
+      const stateData = {
         ...state,
         wallet: action.payload.wallet,
         walletInfo: {
           name: action?.payload?.name,
           icon: action?.payload?.icon,
-          deviceId: action?.payload?.deviceId
+          deviceId: action?.payload?.deviceId,
+          meta: {
+            label: action.payload.meta?.label ?? '',
+            address: (action.payload.wallet as MetaMaskHDWallet | PortisHDWallet).ethAddress ?? ''
```

Comment:
this cast is likely wrong - what about native wallet?

Reply:
From hdwallet typings and implementations, seems like only Portis and Metamask hdwallets have `ethAddress`: 
```shell
❯ rg -g '!node_modules' 'ethAddress'
packages/hdwallet-metamask/src/metamask.ts
53:  ethAddress?: string | null;
212:    if (this.ethAddress) {
213:      return this.ethAddress;
217:      this.ethAddress = address;
220:      this.ethAddress = null;

packages/hdwallet-portis/src/portis.ts
39:  ethAddress?: string;
290:    if (this.ethAddress) return this.ethAddress;
293:    this.ethAddress = out;

packages/hdwallet-keepkey/src/ethereum.ts
189:  const ethAddress = response.proto as Messages.EthereumAddress;
192:  if (ethAddress.hasAddressStr()) address = ethAddress.getAddressStr()!;
193:  else if (ethAddress.hasAddress()) address = "0x" + core.toHexString(ethAddress.getAddress_asU8());
```

I don't know much about the native wallet, but from looking at this PR looks like the wallet name (that should be used in place of an address as identifier according to the issue spec) is coming from `vault.meta`: https://github.com/shapeshift/web/blob/develop/src/context/WalletProvider/NativeWallet/components/NativePassword.tsx#L30 which is why I recommended this typing 🤔 

Looking back into hdwallet though, I'm not sure this will work for Keepkey without any additional implementation.

Reply:
> this cast is likely wrong - what about native wallet?

For native wallet im storing the label and not the address